packages/@react-spectrum/tabs/src/Tabs.tsx

```diff
@@ -34,47 +37,128 @@ export function Tabs<T extends object>(props: SpectrumTabsProps<T>) {
     density,
     ...otherProps
   } = props;
+
   let ref = useRef<HTMLDivElement>();
   let state = useSingleSelectListState<T>({
     ...props,
     onSelectionChange
   });
 
+  let {direction} = useLocale();
   let {styleProps} = useStyleProps(otherProps);
   let {tabListProps, tabPanelProps} = useTabs(props, state, ref);
   let [selectedTab, setSelectedTab] = useState<HTMLElement>();
+  let [collapse, setCollapse] = useState(false);
 
   useEffect(() => {
     let tabs: HTMLElement[] = Array.from(ref.current.querySelectorAll('.' + styles['spectrum-Tabs-item']));
-    setSelectedTab(tabs.find((tab) => tab.dataset.key === state.selectedKey));
-  }, [props.children, state.selectedKey]);
+    let selectedTab = tabs.find((tab) => tab.dataset.key === state.selectedKey);
 
-  return (
-    <div
-      {...styleProps}
-      className={classNames(
-        tabsStyles,
-        'react-spectrum-TabPanel',
-        `react-spectrum-TabPanel--${orientation}`,
-        styleProps.className
-      )}>
+    // default to first tab if selectedTab is deleted from children
+    if (selectedTab != null) {
+      setSelectedTab(selectedTab);
+    } else if (tabs[0]) {
+      state.setSelectedKey(tabs[0].dataset.key);
+      setSelectedTab(tabs[0]);
+    }
+  }, [props.children, state]);
+
+  let onResize = useCallback(() => {
+    let getFurthestPoint = (elem) => {
```

Comment:
does this not work in vertical? i really have no idea if there are designs for this

Reply:
For now, this only addresses horizontal collapse behavior, no designs yet that I found for vertical collapse in the XD file